src/main/scala/chisel3/iotesters/AdvTester.scala

```diff
@@ -98,9 +101,19 @@ abstract class AdvTester[+T <: Module](dut: T,
     expr
   }
 
-  class DecoupledSink[T <: Data, R]( socket: DecoupledIO[T], cvt: T=>R, 
+  class IrrevocableSink[T <: Data, R]( socket: ReadyValidIO[T], cvt: T=>R, 
```

Comment:
Shouldn't  DecoupledSink be kept around with a deprecation instead.  There must be users who will be have change there code if this change goes through